docs/src/service-workers-guide.md

```diff
@@ -0,0 +1,161 @@
+---
+id: service-workers-guide
+title: "Service Workers Guide"
+---
+
+## Disabling Service Workers
+
+[Service Workers](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API) provide a browser-native way to handle requests made with the native [Fetch API (`fetch`)](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API).
+
+If you're using Playwright's [Network-related features](./network.md) like the routing APIs ([`method: BrowserContext.route`], [`method: BrowserContext.routeFromHAR`], [`method: Page.route`], [`method: Page.routeFromHAR`]), or rely heavily on Network Events (e.g. [`event: Page.request`], [`event: BrowserContext.request`], etc.), we recommend disabling Service Workers by setting [`option: Browser.newContext.serviceWorkers`] to `'block'`.
+
+Unless you're testing the Service Worker behavior itself, blocking ensures all requests are routed to your code via Playwright. It also simplifies Network-related APIs within Playwright.
+
+If you need to visibility or routing into all Network events in Firfox or WebKit, currently `'block'`ing Service Workers is required.
+
+## Service Worker Fetch
+
+:::note
+The next sections are only currently supported when using Playwright with Chrome/Chromium.
+:::
+
+### Accessing Service Workers and Waiting for Activation
+
+You can use [`method: BrowserContext.serviceWorkers`] to list the Service Workers, or specifically watch for it:
+
+```js
+const [ serviceWorker ] = await Promise.all([
+  context.waitForEvent('serviceworker'),
+  page.goto('example-with-a-service-worker.html'),
+]);
+```
+
+The [`BrowserContext.serviceworker`] event is emitted ***before*** the Service Worker main script has been evaluated, so before calling [ServiceWorker.evaluate] you should wait on its activation:
+
+```js
+await expect.poll(() => serviceWorker.evaluate(() => self.registration.active?.state)).toBe('activated');
```

Comment:
We probably ought to add an event/PW API for this: something like `serviceWorker.waitForLifecycle('active')`, and/or `serviceWorker.waitForActive()`.

Reply:
can this be achieved using something like this?
```ts
expect.poll(() => page.evaulaute(() => navigator.serviceWorker.controller));
```

Reply:
A variant of that could be used. I was hoping to keep it on the SW PW object instead of the Page, but looks like we don't have generic poll in other language ports, so I've updated the workaround for now.